Rewrite doc test

doc test failed even though docs compiled just fine locally.

I suspect it is because the [action](https://github.com/ammaraskar/sphinx-action) we are using depends on an earlier version of Sphinx (2.4.4)

When I run
```
pip install -Iv sphinx==2.4.4
```
to installed that version, and run `make html` I get build error:
```
Running Sphinx v2.4.4

Extension error:
Event 'html-collect-pages' already present
Makefile:20: recipe for target 'html' failed
make: *** [html] Error 2
```

The action does not seem reliable as it has been more than a year since last update on the repo and it does not have a versioning to ensure stability going into the future. Probably time to revisit the doc check github action so that it doesn't rely on their action.

In any case, I'm sure the action is failing due to a bug in their github action and not our docs

_Originally posted by @UnHumbleBen in https://github.com/UC-Davis-molecular-computing/scadnano-python-package/issues/175#issuecomment-840373419_

--- a/.github/workflows/docs-check.yml
+++ b/.github/workflows/docs-check.yml
@@ -14,9 +14,8 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install sphinx sphinx_rtd_theme
-    - name: Move to docs folder
+    - name: Move to docs folder and build
       run: |
         cd doc
         pwd
-    - name: Build docs
-      run: make html
+        make html

--- a/.github/workflows/docs-check.yml
+++ b/.github/workflows/docs-check.yml
@@ -16,7 +16,7 @@ jobs:
         pip install sphinx sphinx_rtd_theme
     - name: Move to docs folder
       run: |
-        cd docs
+        cd doc
         pwd
     - name: Build docs
       run: make html

--- a/.github/workflows/docs-check.yml
+++ b/.github/workflows/docs-check.yml
@@ -18,4 +18,4 @@ jobs:
       run: |
         cd doc
         pwd
-        make html
+        sphinx-build -T -E -W -b html . _build

--- a/.github/workflows/docs-check.yml
+++ b/.github/workflows/docs-check.yml
@@ -5,9 +5,18 @@ jobs:
   docs:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - uses: ammaraskar/sphinx-action@master
+    - uses: actions/checkout@v2.3.4
+    - name: Set up Python
+      uses: actions/setup-python@v2
       with:
-        docs-folder: "doc/"
-        pre-build-command: "pip3 install sphinx_rtd_theme"
-        build-command: "sphinx-build -T -E -W -b html . _build"
+          python-version: '3.x'
+    - name: Install Sphinx
+      run: |
+        python -m pip install --upgrade pip
+        pip install sphinx sphinx_rtd_theme
+    - name: Move to docs folder
+      run: |
+        cd docs
+        pwd
+    - name: Build docs
+      run: make html

